remoteappmanager/handlers/handler_authenticator.py

```diff
@@ -32,3 +35,23 @@ def authenticate(cls, handler):
                 user = webapp.user
 
         return user
+
+
+class HubOAuthenticator(Authenticator):
+    """Authenticator that uses the remote JupyterHub as an OAuth
+    provider to validate the request."""
+    @classmethod
+    @gen.coroutine
+    def authenticate(cls, handler):
+        # Authenticate the user against the hub. Expects the handler
+        # to inherit from the jupyterhub.services.auth.HubOAuthenticated
+        # mixin
```

Comment:
Since we just use the `get_user` method, theoretically hander just needs to inherit from `HubAuthenticated`